micro-rdk/src/esp32/entry.rs

```diff
@@ -82,6 +83,22 @@ pub async fn serve_web_inner(
         let (cfg_response, cfg_received_datetime) =
             client.get_config(network.get_ip()).await.unwrap();
 
+        if let Some(current_dt) = cfg_received_datetime.as_ref() {
+            let tz = chrono_tz::Tz::UTC;
+            std::env::set_var("TZ", tz.name());
+            let tv_sec = current_dt.timestamp() as i32;
+            let tv_usec = current_dt.timestamp_subsec_micros() as i32;
+            let current_timeval = timeval { tv_sec, tv_usec };
+            let res = unsafe { settimeofday(&current_timeval as *const timeval, std::ptr::null()) };
+            if res != 0 {
+                log::error!(
+                    "could not set time of day for timezone {:?} and timestamp {:?}",
+                    tz.name(),
+                    current_dt
+                );
+            }
+        }
```

Comment:
I'm inclined to keep this because I think it will be a little while before we actually implement time correction. I could check whether the current time of day is the beginning of the epoch before attempting to set the time, would that be acceptable as a stop-gap?

Reply:
That's fair. Could you add a quick comment indicating that this is transitional code until we have a more complete timekeeping story? I don't think trying to decide if the time was already set is going to help, because we can't really know the order in which our code would be brought up vs any timekeeping logic in the containing app.